File: Controllers/DataModel/Sync.cs


using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using lucidity.Controllers;

[Table("Sync")]
public class Sync
{
    public int Id { get; set; }

    [Column("SourceServer_Id")]
    public int SourceServerId { get; set; }

    [ForeignKey("SourceServerId")]
    public Server SourceServer { get; set; }


    [Column("TargetServer_Id")]
    public int TargetServerId { get; set; }

    [ForeignKey("TargetServerId")]

    public Server TargetServer { get; set; }

    [Column("Query_Id")]
    public int QueryId { get; set; }
    public Query Query { get; set; }

    public bool IsScheduleEnabled { get; set; }

    public string CronSchedule { get; set; }

    [Timestamp]
    [ConcurrencyCheck]
    public byte[] RowVersion { get; set; }
}

File: client/app/models/Sync.ts
import { Server } from "./Server";
import { Query } from "./Query";

export class Sync {

    constructor() {
        this.id = 0;
        this.sourceServer =  null;
        this.sourceServerId = 0;
        this.targetServer = null;
        this.targetServerId = 0;
        this.query = null;
        this.queryId = 0;
        this.isScheduleEnabled = false;
        this.cronSchedule = '';
        this.rowVersion = null;
    }

    get name() {
        let temp = '' + this.id;
        if (!!this.query)
            return temp + ' ' + this.query.name;
        return temp;
    }

    id: number;

    sourceServer: Server;
    sourceServerId: number;

    targetServer: Server;
    targetServerId: number;

    query: Query;
    queryId: number;

    isScheduleEnabled: boolean;
    cronSchedule: string;

    rowVersion: any;
}
